src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -47139,6 +47130,1576 @@ void M5(object? x)
+    {
+        _ = c?.M0(out x) == c!.M0(out y)
+            ? x.ToString() + y.ToString() // 1
+            : x.ToString() + y.ToString(); // 2
+    }
+
+    public void M2(C? c, object? x, object? y)
+    {
+        _ = c?.M0(out x) != c!.M0(out y)
+            ? x.ToString() + y.ToString() // 3
+            : x.ToString() + y.ToString(); // 4
+    }
+
+    public void M3(C? c, object? x, object? y)
+    {
+        _ = c!.M0(out x) == c?.M0(out y)
+            ? x.ToString() + y.ToString() // 5
+            : x.ToString() + y.ToString(); // 6
+    }
+
+    public void M4(C? c, object? x, object? y)
+    {
+        _ = c!.M0(out x) != c?.M0(out y)
+            ? x.ToString() + y.ToString() // 7
+            : x.ToString() + y.ToString(); // 8
+    }
+}
+";
+            CreateNullableCompilation(source).VerifyDiagnostics(
+                // (9,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() + y.ToString() // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(9, 15),
+                // (10,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString() + y.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(10, 15),
+                // (16,15): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() + y.ToString() // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(16, 15),
+                // (17,15): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString() + y.ToString(); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x").WithLocation(17, 15),
+                // (23,30): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() + y.ToString() // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y").WithLocation(23, 30),
+                // (24,30): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString() + y.ToString(); // 6
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y").WithLocation(24, 30),
+                // (30,30): warning CS8602: Dereference of a possibly null reference.
+                //             ? x.ToString() + y.ToString() // 7
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y").WithLocation(30, 30),
+                // (31,30): warning CS8602: Dereference of a possibly null reference.
+                //             : x.ToString() + y.ToString(); // 8
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "y").WithLocation(31, 30)
+                );
+        }
+
+        /// <remarks>Ported from <see cref="FlowTests.EqualsCondAccess_04"/>.</remarks>
+        [Fact]
+        public void EqualsCondAccess_04_03()
```

Comment:
I think we should also have a test like `if (a?.b(out x) == M0(y = x)) y.ToString();`